Source/DafnyLanguageServer/Language/Symbols/SymbolTable.cs

```diff
@@ -62,7 +62,13 @@ bool symbolsResolved
     /// <returns><c>true</c> if a symbol was found, otherwise <c>false</c>.</returns>
     /// <exception cref="System.InvalidOperationException">Thrown if there was one more symbol at the specified position. This should never happen, unless there was an error.</exception>
     public bool TryGetSymbolAt(Position position, [NotNullWhen(true)] out ILocalizableSymbol? symbol) {
-      symbol = LookupTree.Query(position).SingleOrDefault();
+      var symbolsAtPosition = LookupTree.Query(position);
+      // Use case: function f(a: int) {}, and hover over a.
+      if (symbolsAtPosition.Count() > 1) {
```

Comment:
Note: See my comment in `SymbolTableFactory`.

Careful, the `IEnumerable` interface does not guarantee that the instance can be enumerated multiple times.
Moreover, the if/else statement doesn't provide any benefit. If you don't want the `TryGetSymbolAt` method to fail if there are multiple entries at the same location, use `FirstOrDefault`.
However, I stand by the point that `FirstOrDefault` will hide implementation bugs. If a query mustn't return more than one result, the code should express this. The `LookupTree`, mustn't contain entries that overlap with each other (or duplicates).
PS: I generally don't like this kind of "fail-safety" since it's hard to debug, and errors might only surface late in the pipeline, if at all. Therefore, I'd always prefer the fail-fast approach to catch errors as early as possible. Moreover, this error is not logged here (noting that logs are generally overseen if the application appears to work).

Reply:
The if/else statement provided the debugging benefit. I'm developing in Dafny while there is a breakpoint here, so whenever this happens, I can quickly find what is the faulty symbol.

Among the all the possible classification of errors, I think we agree that having duplicates in the lookup tree should only be a developer-facing warning, not a user-facing crash, because it not going to cause any user-facing errors, since the only place where we pick up something from the lookup tree is this very code.

I'll change the code so that 1) there is at most one enumeration. and 2) It logs as a warning if there is a duplicate, so that I can use 2) as a breakpoint if I detect other errors.

We should definitely use Dafny at some point so that we can prove properties like what we assume on the code :-)